tests/FilterAttachedResults/FilterAttachedResults.spec.ts

```diff
@@ -0,0 +1,55 @@
+import {
+    FilterAttachedResults,
+    IFilterAttachedResultsOptions
+} from '../../src/components/FilterAttachedResults/FilterAttachedResults';
+import { Mock, Simulate } from 'coveo-search-ui-tests';
+
+describe('FilterAttachedResults', () => {
+    let filter: Mock.IBasicComponentSetup<FilterAttachedResults>;
+
+    beforeEach((done) => {
+        filter = Mock.basicResultComponentSetup<FilterAttachedResults>(
+            FilterAttachedResults
+        );
+        setTimeout(done, 0);
+    });
+
+    afterEach(() => {
+        filter = null;
+    });
+
+    it("should have the required elements", () => {
+        expect(filter.cmp.element.childNodes.length).toBe(1);
+        const mainSection = filter.cmp.element.childNodes[0];
+        expect(filter.cmp.element.querySelector('.CoveoFacet')).not.toBeNull();
+        expect(mainSection.childNodes.length).toBe(1);
+        expect(mainSection.firstChild.childNodes.length).toBe(1);
+        expect(filter.cmp.element.querySelector('input[type=checkbox]')).not.toBeNull();
+    });
+
+    describe("when setting options", () => {
+        beforeEach((done) => {
+            filter = Mock.optionsComponentSetup<FilterAttachedResults, IFilterAttachedResultsOptions>(
+                FilterAttachedResults,
+                { text: "much rain", getUriHashes: () => ["foo", "bar"] }
+            );
+            setTimeout(done, 0);
```

Comment:
Can you use clock instead of direcxt setTimeout ?
See https://jasmine.github.io/api/2.6/Clock.html.